src/training.rs

```diff
@@ -329,12 +354,14 @@ impl<B: Backend> FSRS<B> {
         Ok(optimized_parameters)
     }
 
-    pub fn benchmark(
-        &self,
-        train_set: Vec<FSRSItem>,
-        enable_short_term: bool,
-        num_relearning_steps: Option<usize>,
-    ) -> Vec<f32> {
+    pub fn benchmark(&self, input: ComputeParametersInput) -> Vec<f32> {
+        let ComputeParametersInput {
+            train_set,
+            enable_short_term,
+            num_relearning_steps,
+            ..
+        } = input;
+
```

Comment:
```suggestion
    pub fn benchmark(&self, ComputeParametersInput {
            train_set,
            enable_short_term,
            num_relearning_steps,
            ..
        } : ComputeParametersInput) -> Vec<f32> {
```

I guess this will be shorter.

other places as well.